sklearn/model_selection/_validation.py

```diff
@@ -484,6 +484,14 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
         score_time = 0.0
         if error_score == 'raise':
             raise
+        elif error_score == 'raise-deprecating':
+            warnings.warn("Estimator failed to fit for these parameters; the "
+                          "error will be raised. If you want a value to be "
+                          "assigned to the score in such cases, then specify "
+                          "error_score parameter. Default is 'raise' but from "
+                          "version 0.22 it will change to np.nan.",
```

Comment:
I think this warning should be more clear a) about why a user is getting it b) how to make it go away. 
Something along the lines of what @jnothman said in https://github.com/scikit-learn/scikit-learn/pull/10677#discussion_r170446753  might be better I think,

>  "From version 0.22, errors during fit will result in a cross validation score of NaN by default. Use error_score='raise' if you want an exception raised."

Once sentence for a) and one for b) . and maybe add  "[...]  or error_score=np.nan to adopt the behavior from version 0.22."